fulp_modules/lisa/joel_gun.dm

```diff
@@ -0,0 +1,195 @@
+			 "<span class='userdanger'>You stare directly into the souls of those around you.</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/H in viewers(1, user))
+				if(prob(75) && !(H == user) && !(H.stat))
+					to_chat(H, "<span class='warning'>You feel terrifyingly spooked by [user.name]!</span>")
+					if(!H.mind || !istype(H.mind.martial_art, /datum/martial_art/velvetfu))
+						var/throwtarget = get_edge_target_turf(user, get_dir(user, get_step_away(H, user)))
+						H.throw_at(throwtarget, 3, 2)
+			used_ability = TRUE
+			addtimer(CALLBACK(src, .proc/clear_cooldown), 10 SECONDS)
+			return
+	else
+		..()
+		spin()
+
+/datum/voice_of_god_command/repulse/execute(list/listeners, mob/living/user, power_multiplier = 1, message)
+	for(var/mob/living/target as anything in listeners)
+		var/throwtarget = get_edge_target_turf(user, get_dir(user, get_step_away(target, user)))
+		target.throw_at(throwtarget, 3 * power_multiplier, 1 * power_multiplier)
+
+/// Steady Aim
+/obj/item/gun/ballistic/revolver/joel/attack_secondary(mob/living/victim, mob/living/user, params)
+	if(used_ability)
+		to_chat(user, "<span class='warning'>You have to wait before using an ability!</span>")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(user == victim)
+		to_chat(user,"<span class='warning'>You can't steady aim yourself!</span>")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	user.visible_message("<span class='danger'>[user.name] holds [victim.name] at gunpoint!</span>",\
+	 "<span class='userdanger'>You hold [victim.name] up at gunpoint!</span>",\
+	 "<span class='hear'>You hear metal clanking...</span>")
+	playsound(loc, 'fulp_modules/lisa/Sounds/steadyaim.ogg', 50, FALSE, -5)
+	if(prob(50))
+		if(!victim.mind || !istype(victim.mind.martial_art, /datum/martial_art/velvetfu))
+			victim.Stun(rand(10, 20))
+	used_ability = TRUE
+	addtimer(CALLBACK(src, .proc/clear_cooldown), 8 SECONDS)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/// Prevents the gun from being fired.
+/obj/item/gun/ballistic/revolver/joel/afterattack(atom/target, mob/living/user, flag, params)
+	if(!chambered) /// Can shoot if its empty, we don't care!
+		user.visible_message("<span class='danger'>*click*</span>")
+		playsound(loc, 'fulp_modules/lisa/Sounds/gunclick.ogg', 50, FALSE, -5)
+		return
+	if(used_ability)
+		to_chat(user, "<span class='warning'>You don't want to waste [src]'s only bullet!</span>")
+		return
+	if(target) /// Misdirection (Scholar of the Wilbur Sin exclusive)
+		user.visible_message("<span class='danger'>[user.name] looks around themselves.</span>",\
+		 "<span class='userdanger'>You look for something to point [src] at.</span>")
+		playsound(loc, 'fulp_modules/lisa/Sounds/misdirect.ogg', 50, FALSE, -5)
+		used_ability = TRUE
+		addtimer(CALLBACK(src, .proc/clear_cooldown), 7 SECONDS)
+		if(do_after(user, 0.4 SECONDS, target = src, progress = FALSE))
```

Comment:
why is this do_after longer than the others by a tenth of a second?

Reply:
So it goes with the sound effect playing